api/v1beta2/ibmvpccluster_types.go

```diff
@@ -103,25 +114,118 @@ type VPCLoadBalancerStatus struct {
-	VPC VPC `json:"vpc,omitempty"`
 
-	// Ready is true when the provider resource is ready.
+	// Conditions defines current service state of the load balancer.
 	// +optional
-	Ready       bool        `json:"ready"`
-	Subnet      Subnet      `json:"subnet,omitempty"`
-	VPCEndpoint VPCEndpoint `json:"vpcEndpoint,omitempty"`
+	Conditions capiv1beta1.Conditions `json:"conditions,omitempty"`
 
 	// ControlPlaneLoadBalancerState is the status of the load balancer.
+	// dep: rely on NetworkStatus instead.
 	// +optional
 	ControlPlaneLoadBalancerState VPCLoadBalancerState `json:"controlPlaneLoadBalancerState,omitempty"`
 
-	// Conditions defines current service state of the load balancer.
+	// COSInstance is the reference to the IBM Cloud COS Instance used for the cluster.
+	COSInstance *ResourceReference `json:"cosInstance,omitempty"`
+
+	// networkStatus is the status of the VPC network in its entirety resources.
+	NetworkStatus *VPCNetworkStatus `json:"networkStatus,omitempty"`
+
+	// ready is true when the provider resource is ready.
+	// +kubebuilder:default=false
+	Ready bool `json:"ready"`
+
+	// resourceGroup is the reference to the IBM Cloud VPC resource group under which the resources will be created.
+	ResourceGroup *ResourceReference `json:"resourceGroupID,omitempty"`
+
+	// dep: rely on NetworkStatus instead.
+	Subnet Subnet `json:"subnet,omitempty"`
+
+	// dep: rely on NetworkStatus instead.
+	VPC VPC `json:"vpc,omitempty"`
+
+	// dep: rely on ControlPlaneEndpoint
+	VPCEndpoint VPCEndpoint `json:"vpcEndpoint,omitempty"`
+}
+
+// VPCNetworkStatus provides details on the status of VPC network resources.
+type VPCNetworkStatus struct {
+	// computeSubnets references the VPC Subnets for the cluster's Data Plane.
 	// +optional
-	Conditions capiv1beta1.Conditions `json:"conditions,omitempty"`
+	ComputeSubnets []*VPCResourceStatus `json:"computeSubnets,omitempty"`
+
+	// controlPlaneSubnets references the VPC Subnets for the cluster's Control Plane.
+	// +optional
+	ControlPlaneSubnets []*VPCResourceStatus `json:"controlPlaneSubnets,omitempty"`
+
+	// loadBalancers references the VPC Load Balancer's for the cluster.
+	// +optional
+	LoadBalancers []VPCLoadBalancerStatus `json:"loadBalancers,omitempty"`
+
+	// publicGateways references the VPC Public Gateways for the cluster.
+	// +optional
+	PublicGateways []*VPCResourceStatus `json:"publicGateways,omitempty"`
+
+	// securityGroups references the VPC Security Groups for the cluster.
+	// +optional
+	SecurityGroups []*VPCResourceStatus `json:"securityGroups,omitempty"`
+
+	// vpc references the IBM Cloud VPC.
+	// +optional
+	VPC *VPCResourceStatus `json:"vpc,omitempty"`
+}
+
+// VPCResourceStatus identifies a resource by crn and type and whether it was created by the controller.
+type VPCResourceStatus struct {
+	// controllerCreated indicates whether the resource is created by the CAPI controller.
+	ControllerCreated bool `json:"controllerCreated,omitempty"`
```

Comment:
Since most of VPC resources can be tagged, better to make use of tags instead of ControllerCreated. tags will be very easy comapared to managing state of this variable.

Reply:
I may need to think on this one more. I could move to rely on tags, but I know in the past we could not, or simply did not, in terms of performing resource management, perhaps a limitation with VPC resources at the time, or the technology/component used to create such resources.
Perhaps CAPI relies on tags information, if possible, and outside layers and logic transform that information as necessary. I may not know more on how to handle this until I start getting deeper into interdependencies, expectancies, etc.

Do you have a suggestion for this, or simply added a `Tags []string ...`? I didn't see a PowerVS equivalent, and I think the `Status` results will be necessary for post-Infrastructure-provisioning processing.

Reply:
Or are you expecting that CAPI (or other components/automation) would look for specific Tags on resources (say a VSI with `CAPI-created` tag or something like that)?

Reply:
I dropped the bool for now.

Reply:
Apologies some how I missed it. Yes what I am thinking is once the capi creates the resource in cloud say VPC, lets tag that resource with tag like CAPI-created or something, so it would help us in deciding on deleting that resource.

In PowerVS we cannot make use of tags as few services doesnot support tagging like DHCP, Network so on